src/w_wad.h

```diff
@@ -22,6 +22,15 @@
 #pragma interface
 #endif
 
+// a raw entry of the wad directory
+// NOTE: This sits here and not in w_wad.c because p_setup.c makes use of it to load map WADs inside PK3s.
+typedef struct
+{
+	UINT32 filepos; // file offset of the resource
+	UINT32 size; // size of the resource
+	char name[8]; // name of the resource
+} ATTRPACK filelump_t;
```

Comment:
This will break on MSVC
you need 
```#if defined(_MSC_VER)
#pragma pack(1)
#endif` and `#if defined(_MSC_VER)
#pragma pack()
#endif` to tell MSVC that the the struct is packed to 1 byte

Reply:
Done.